compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -167,6 +167,22 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
   private def arrayToSeq(tree: Tree)(using Context): Tree =
     wrapArray(tree, tree.tpe.elemType)
 
+  /**
+    * Wrap vararg splices that have been lifted to valdefs in a call to `arrayToSeq`.
+    * `SeqLiteral`s are not wrapped, since they are eliminated in their own phase see `SeqLiterals`.
+    */
+  override def transformValDef(tree: ValDef)(using Context): Tree =
+    if tree.tpt.typeOpt.isRepeatedParam then
```

Comment:
That doesn't seem right. An isolated `ValDef` should never have a repeated type in the first place. IMO the problem is earlier, located at the point where we create that `ValDef` in the first place.

Reply:
At first, I thought that this either shouldn't be lifted or should be handled when lifting, but I found some commits that explicitly fix lifted repeated params in this phase e.g. https://github.com/lampepfl/dotty/pull/15530
Do you think that this should be handled when lifting? Or just not lifted at all?

Reply:
Meh, it seems to me that handling those in `ElimRepeated` is a workaround.

IMO the `Typed` node turning a `Seq` into a `*` type belongs close to the corresponding `Apply`. In a sense, the combination `Apply+Typed`-inside has a specific meaning, so we should not separate a `Typed` node for `*` from its `Apply` node.

Of course, we can still lift the expression *inside* the `Typed` node. That is probably what the coverage phase should do.